server/src/main/java/org/elasticsearch/cluster/SnapshotDeletionsInProgress.java

```diff
@@ -167,29 +173,35 @@ public String toString() {
      * A class representing a snapshot deletion request entry in the cluster state.
      */
     public static final class Entry implements Writeable, RepositoryOperation {
-        private final Snapshot snapshot;
+        private final List<SnapshotId> snapshots;
```

Comment:
why did you opt for having one entry for multiple snapshots? Should we rather have one entry per snapshot that is to be deleted?

If we build it that way, could it allow us to queue up multiple deletes in the future?

Reply:
> why did you opt for having one entry for multiple snapshots? 

I thought about that actually. What I didn't like about it, was that it introduced quite a bit of duplication in what is stored in the cluster state. We'd be adding a bunch of deletes with the same start time, repository and generation to then batch them up again when it comes to running the actual delete.

> If we build it that way, could it allow us to queue up multiple deletes in the future?

I don't think that makes things easier for queuing up deletes. My thinking is this:

A delete will always move us from a set of snapshots to another set of snapshots and move the repository generation ahead by `1`. It has a defined point in time where it starts (delete entry is processed in the cluster state) and a defined repository state from which it starts. We cannot add to the delete while it is running against the repo as that's an atomic operation also.

So if we were to add deletes by just adding entries, then we'll have to queue up new deletes at a generation higher than the lowest generation of existing deletes (the lowest generation is already processing in the repository so we can't add to it).
=> I figured I'd rather be explicit about how deletes get batched together than do some implicit magic around the repository generations here.
With the way it is now, we can queue up deletes very nicely IMO: If there is only a single delete entry in the CS, we add a new entry because that one is already processing on the repo. If there are multiple delete entries we can add the additional snapshots we want to delete to the second one (that we know isn't processing yet).

Reply:
ok, makes sense